Sources/Workspace/Workspace+Configuration.swift

```diff
@@ -260,53 +260,38 @@ extension Workspace.Configuration {
+                #else
+                throw InternalError("Keychain not supported on this platform")
+                #endif
+            case .disabled:
+                // noop
+                break
+            }
 
             switch self.netrc {
             case .custom(let path):
                 guard fileSystem.exists(path) else {
                     throw StringError("Did not find .netrc file at \(path).")
                 }
                 providers.append(try NetrcAuthorizationProvider(path: path, fileSystem: fileSystem))
-            case .workspaceAndUser(let rootPath):
-                // package/project "local" .netrc file, takes priority over user-level file
-                let localPath = rootPath.appending(component: ".netrc")
-                // user didn't tell us to explicitly use these .netrc files so be more lenient with errors
-                if let localProvider = self.loadOptionalNetrc(fileSystem: fileSystem, path: localPath, observabilityScope: observabilityScope) {
-                    providers.append(localProvider)
-                }
-
-                // user .netrc file (most typical)
-                let userHomePath = try fileSystem.homeDirectory.appending(component: ".netrc")
-                // user didn't tell us to explicitly use these .netrc files so be more lenient with errors
-                if let userHomeProvider = self.loadOptionalNetrc(fileSystem: fileSystem, path: userHomePath, observabilityScope: observabilityScope) {
-                    providers.append(userHomeProvider)
-                }
             case .user:
                 // user .netrc file (most typical)
                 let userHomePath = try fileSystem.homeDirectory.appending(component: ".netrc")
 
                 // user didn't tell us to explicitly use these .netrc files so be more lenient with errors
-                if let userHomeProvider = self.loadOptionalNetrc(fileSystem: fileSystem, path: userHomePath, observabilityScope: observabilityScope) {
+                if let userHomeProvider = try? NetrcAuthorizationProvider(path: userHomePath, fileSystem: fileSystem) {
                     providers.append(userHomeProvider)
                 }
-            case .disabled:
-                // noop
-                break
             }
 
-            switch self.keychain {
-            case .enabled:
-                #if canImport(Security)
-                providers.append(KeychainAuthorizationProvider(observabilityScope: observabilityScope))
-                #else
-                throw InternalError("Keychain not supported on this platform")
-                #endif
-            case .disabled:
-                // noop
-                break
-            }
-
-            return providers.isEmpty ? .none : CompositeAuthorizationProvider(providers, observabilityScope: observabilityScope)
+            // Use at-most one AuthorizationProvider (i.e., no CompositeAuthorizationProvider)
```

Comment:
👀

Reply:
this could be an issue for users that rely on netrc for example for binary dependencies, so we may need to add an option allowing macOS users to "force netrc mode" as they transition

Reply:
cc @neonichu for opinions

Reply:
added `forceNetrc` option